apps/web/src/services/siwe/useSiwe.tsx

```diff
@@ -0,0 +1,53 @@
+import { useWeb3 } from '@/hooks/wallets/web3'
+import { useAuthGetNonceV1Query, useAuthVerifyV1Mutation } from '@safe-global/store/gateway/AUTO_GENERATED/auth'
+import { useCallback } from 'react'
+
+export const useSiwe = () => {
+  const provider = useWeb3()
+  const { refetch: fetchNonce } = useAuthGetNonceV1Query()
+  const [verifyAuthMutation] = useAuthVerifyV1Mutation()
+
+  const getSignableMessage = (address: string, chainId: bigint, nonce: string) => {
```

Comment:
I would suggest moving this outside of the hook into a `utils` file in the same folder.